plutus-contract/src/Plutus/Contract/Test.hs

```diff
@@ -302,12 +301,19 @@ checkPredicateInnerStream CheckOptions{_minLogLevel, _emulatorConfig}
               theStream
 
     case result of
-      Right (True  S.:> res) -> return res
+      Right (True  S.:> res) -> do
+        -- Also log on success. The test runner suppresses the output on success so this has no effect on succesful tests.
+        -- But this makes `checkPredicateInnerStream` more modular: When another test function calls it,
+        -- and that function has a failure, the logs will be shown.
+        _ <- logEmulator
```

Comment:
I tend to prefer `void logEmulator` over `_ <- logEmulator`. If it's fine for you, you can update it in the other branches as well.